_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+
+### Create an ECR Repository
+
+First, we need to make one final change to the **Dockerfile** to get TLS working on Fargate.
+
+Add the following two lines to the **Dockerfile** right after the `dotnet publish` command:
+
+```dockerfile
+# build and publish the app
+...
+RUN dotnet publish -c release -o /app --no-restore
+
+ARG CERT_PASSWORD
+RUN dotnet dev-certs https -ep /app/aspnetapp.pfx -p ${CERT_PASSWORD}
+
+# final stage/image
+...
+```
+
+This will create a self-signed development certificate right within the Docker image, which we can use to run the application with TLS on Fargate for demo purposes.
+
+> **Note**: Using these development certificates is not recommended for production use. I'm using it here for simplicity as this is just a demo. For production setup, you should use a certificate signed by a certificate authority. The recommended way to run a production .NET application on ECS would be to use an AWS Application Load Balancer (ALB) to route traffic to a reverse proxy (Nginx) via an HTTPS listener configured to use a certificate signed by a certificate authority. The reverse proxy would then route traffic to the application. The reverse proxy and the application should also be configured to use TLS using valid certificates.
+
+Login to your AWS console and navigate to ECR.
+
+Click **Create Repository**.
+
+For visibility settings, choose **Private** and name your repository. In my case, I'm going with _okta-chat_. Then click **Create Repository** at the bottom of the wizard.
+
+{% img blog/dotnet-aws-ecs-fargate/16-reponaming.png alt:"AWS Create repo" width:"800" %}{: .center-image }
+
+Navigate to **okta-chat** and click **View push commands** this has all the steps you'll need to build and push your image to ECR.
+
+{% img blog/dotnet-aws-ecs-fargate/17-PushCommands.png alt:"Push commands" width:"800" %}{: .center-image }
+
+Follow the steps and push the images to ECR. Make sure to pass `--build-arg CERT_PASSWORD=mypass123` to the `docker build` command like below.
+
+```bash
+docker build --build-arg CERT_PASSWORD=mypass123 -t okta-chat .
+```
+
+Select the copy button in the URI column next to your image and save that for later. This is the path to your image.
+
+{% img blog/dotnet-aws-ecs-fargate/18-oktaChatRepo.png alt:"AWS save URI" width:"800" %}{: .center-image }
+
+### Set up the ECS Fargate Cluster
+
+Now set up your ECS cluster.
+
+On the left menu, click **Clusters** under **Amazon ECS**, then click the **Create Cluster** button.
+
+Click **Networking Only** > **Next step**from the options.
+
+{% img blog/dotnet-aws-ecs-fargate/19-NetworkOnly.PNG alt:"AWS Networking Only" width:"800" %}{: .center-image }
+
+I named my cluster "okta-test", name yours as you see fit, and click **Create** on the following screen click **View Cluster**.
+
+Now that the cluster is set-up you'll need to create a task for your image to run as.
```

Comment:
```suggestion
Now that the cluster is set up you'll need to create a task for your image.
```
style guide. i don't know how to rewrite "to run as" but it's ambiguous so i dropped it